### PR TITLE
feat(envs): dynamic variables for envs

### PR DESCRIPTION
**Pull request by michael-0acf4**
Enable support for dynamic envs,  and solves #80 
The function provided will be evaluated within a task.
```ts
env("main")
  .var("A", "A#STATIC")
  .var("C", ($) => $`echo C [$A, $B]`.text())
  .var("B", () => "B#DYNAMIC")
  .onEnter(task(($) => $`echo $A, $B, $C`)); 
  // will print "A#STATIC, B#DYNAMIC, C [A#STATIC, B#DYNAMIC]" 
```

#### Migration notes

None

- [ ] The change comes with new or modified tests
- [ ] Hard-to-understand functions have explanatory comments
- [ ] End-user documentation is updated to reflect the change

### CODE REVIEW

File: modules/envs/reducer.ts [hunk 1]
```diff
@@ -85,3 +87,17 @@ export async function reduceStrangeProvisions(
...
+
+function getEnvReducer(_gcx: GhjkCtx) {
+  // TODO:
+  // How to exec task from here? how to look for envs.#task?
+  // execTask(gcx, tasksConfig, taskGraph, targetKey, args)
+  // await execTask(gcx, {...??}, {}, "", "");
+  return (provisions: Provision[]) => {
+    return Promise.resolve(provisions.map((p) => {
+      p.ty = "posix.envVar";
+      // TODO
```
Comment (afmika) [inline modules/envs/reducer.ts:99]: @Yohe-Am on the task approach, not really sure how I can execute the task from the reducer, the key is known at this point as `p.val`.. not sure if I am approaching it the wrong way.

**Review by destifo — APPROVED** [review]
Nice! can you update the PR note

File: examples/env_vars/ghjk.ts [hunk 2]
```diff
@@ -0,0 +1,21 @@
+import { file } from "../../mod.ts";
+
+const ghjk = file({
```
Comment (Yohe-Am) [inline examples/env_vars/ghjk.ts:3]: nitpick: I'd use `hack.ts` for this example.

File: modules/ports/mod.ts [hunk 3]
```diff
@@ -108,7 +112,10 @@ export class PortsModule extends ModuleBase<PortsCtx, PortsLockEnt> {
       installSetProvisionTy,
       installSetReducer(gcx) as ProvisionReducer<Provision, Provision>,
     );
-
+    reducerStore.set(
```
Comment (Yohe-Am) [inline modules/ports/mod.ts:115]: Any reason why you picked the ports module to house this? The reduce doesn't use any ports features, does it?

File: deps/common.ts [hunk 4]
```diff
@@ -11,6 +11,7 @@ export * as std_url from "https://deno.land/std@0.213.0/url/mod.ts";
 export * as std_path from "https://deno.land/std@0.213.0/path/mod.ts";
 export * as std_fs from "https://deno.land/std@0.213.0/fs/mod.ts";
 export * as zod_val_err from "npm:zod-validation-error@3.3.0";
+export { default as hashAnyValue } from "https://deno.land/x/object_hash@2.0.3.1/mod.ts";
```
Comment (Yohe-Am) [inline deps/common.ts:14]: Couldn't find where you used this? The hashing functions already in `utils/mod.ts` do a lot of things in addition to hashing. I'd stick to those or build on top of them.

**Review by Yohe-Am — CHANGES_REQUESTED** [review]
Looking good. Just needs tests.

File: modules/ports/mod.ts (hunk 3 above)
Comment (afmika) [reply modules/ports/mod.ts:115]: No real reasons, I put it there as this seemed to be the only place I reducers were set. Initially, I put it [here](https://github.com/metatypedev/ghjk/blob/met-531-dynamic-variables-for-envs/modules/envs/reducer.ts#L31)

File: deps/common.ts (hunk 4 above)
Comment (afmika) [reply deps/common.ts:14]: Sorry, this was a left-over as I tried to hash functions on a previous attempt.

**Review by Yohe-Am — APPROVED** [review]
Nice. I'd merge the #96 and rebase on top of that to fix issues not related to this PR.

**Review by Natoandro — APPROVED** [review]

**Pull request merged**